ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -376,7 +376,8 @@ private void recover(MessageStoreRecovery recovery) throws StoreException, IOExc
           IndexValue newValue = new IndexValue(info.getSize(), runningOffset, IndexValue.FLAGS_DEFAULT_VALUE,
               info.getExpirationTimeInMs(), info.getOperationTimeMs(), info.getAccountId(), info.getContainerId(),
               info.getLifeVersion());
-          addToIndex(new IndexEntry(info.getStoreKey(), newValue, null), new FileSpan(runningOffset, infoEndOffset));
+          addToIndex(new IndexEntry(info.getStoreKey(), newValue, null), new FileSpan(runningOffset, infoEndOffset),
+              true);
```

Comment:
I don't understand this? Why is this the only place that we set the last parameter to true? In this entire `recover` method, all the `IndexEntry` added to the persistent index should have the last parameter set to true. Like line 336, line 344, line 360. 
The problem is that those method `markAsDeleted`, `markAsUndelete` and `markAsPermanent` has another wrapper method around `addToIndex`, so it would be hard to change all these methods. So I suggest that we create a new object field `inRecovering` and set it true when starting recovering and set it false when finishing recovering.

Reply:
Sure. updated it and move the clean up journal method outside of recovery.